main.py

```diff
@@ -88,6 +89,17 @@ def knapsack_bqm(costs, weights, weight_capacity):
 
 sampler = LeapHybridSampler()
 sampleset = sampler.sample(bqm)
-solution = [df['weight'][i] for i in range(len(df['weight'])) if sampleset.record.sample[0][i] == 1.]
-
-print("Found solution {} at energy {}.".format(solution, sampleset.first.energy))
+n_sols = sampleset.record.sample.shape[0]
+for i in range(n_sols):
```

Comment:
Rather than looping by indexing, it would be more pythonic to do the following:
```
for sample in sampleset.record.sample:
    ..
```

Reply:
As a follow up, why are we looping through all the samples in the sampleset?

Don't we only care about the lowest energy sample? If so, shouldn't we simply be using `sampleset.lowest().record.sample`?

Reply:
@m3ller, I changed the code - thanks. I needed the energy, too, so I did zip(), as you will see.

About the lowest - I didn't want to make that assumption, if the solver is returning more than one.